src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,162 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T[] array;
+    private int lastElementIndex;
+
+    public ArrayList() {
+        array = (T[])new Object[10];
```

Comment:
10 is magic number. 
should be extracted into constant.
```suggestion
        array = (T[])new Object[];
```